nix/ocaml.nix

```diff
@@ -85,16 +91,150 @@ let
     [ zlib bzip2 gmp openssl libffi ]
     ++ lib.optional (!(stdenv.isDarwin && stdenv.isAarch64)) jemalloc;
 
+  dune-nix = inputs.dune-nix.lib.${pkgs.system};
+
+  base-libs = dune-nix.squashOpamNixDeps scope.ocaml.version
```

Comment:
What happens when someone changes `opam.export`? Does `dune-nix` re-generate some nix parts, or nix reads `opam.export` every time?

From looking at `dune-description` and how it's used in `duneDescLoaded` and `info`, it seems the latter?